src/main/java/net/syscon/elite/api/model/ImprisonmentStatus.java

```diff
@@ -0,0 +1,20 @@
+package net.syscon.elite.api.model;
+
+import lombok.Data;
+
+@Data
+public class ImprisonmentStatus {
+
+    private Long bookingId;
+    private String bandCode;
+    private String imprisonmentStatus;
+    private String legalStatus;
+
+    public void deriveLegalStatus() {
```

Comment:
Just a thought (as you are relying on this being called).  Can you just not put in the getter (so replacing th@Data genereted on) so it derives up calling?